control-plane/api/v1alpha1/shared_types.go

```diff
@@ -16,6 +16,8 @@ import (
 
 // This file contains structs that are shared between multiple config entries.
 
+const metaValueMaxLength = 512
```

Comment:
Is this max length in bytes? maybe we can add the units? Also is it a common type it's only used by the intentions right?

Reply:
It is the max length of the string size. In Consul it is used by more than just intention meta, any config entry meta should not be more than this length so I added it to shared types to later use in validating other config entries. 
I will add test coverage for this check.

Reply:
ah okay, that's great then. Can you add a comment to that effect? Thanks!